src/com/android/volley/RequestQueue.java

```diff
@@ -301,7 +295,16 @@ public boolean isProcessing(Request< ? > request) {
                 mWaitingRequests.put(cacheKey, null);
             }
 
-            mCacheQueue.add(request);
+            if (!request.isJoined()) {
```

Comment:
can the request not join if it is in fact network-only? if not, this will prevent network-only requests from actually going to the network afaict